src/engine/ruleset/decoders/fortinet/fortimail.yml

```diff
@@ -0,0 +1,199 @@
+        type=<event.action> subtype=<~log.rsa.internal.messageid> pri=<log.level> msg="<~log.rsa.internal.event_desc>"
+
+normalize:
+  - map:
+    - event.dataset: fortinet.fortimail
+    - event.module": fortinet
+    - fileset.name": fortimail
+    - input.type": log
+    - observer.type: Firewall
+    - observer.vendor: Fortinet
+    - observer.product: FortiMail
+    - rsa.misc.reference_id1: $~log.rsa.misc.reference_id1
+    - rsa.misc.reference_id: $event.code
+    - rsa.misc.msgIdPart1: $event.action
+    - rsa.misc.msgIdPart2: $~log.rsa.internal.messageid
+    - rsa.internal.messageid: $event.action
+    - rsa.internal.messageid: $~log.rsa.internal.messageid
+    - rsa.misc.event_type: $event.action
+    - rsa.misc.severity: $log.level
+    - rsa.misc.filter: $~log.rsa.misc.filter
+    - rsa.misc.result: $~log.rsa.misc.result
+    - rsa.network.domain: $server.domain
+    - rsa.misc.log_session_id: $~log.rsa.misc.filter
+    - rsa.email.email_src: $~log.rsa.email.email_src
+    - rsa.misc.client: $~log.rsa.misc.client
+    - rsa.web.fqdn: $~log.rsa.web.fqdn
+    - rsa.misc.context: $~log.rsa.misc.context
+    - rsa.email.email_dst: $~log.rsa.email.email_dst
+    - rsa.misc.virusname: $~log.rsa.misc.virusname
+    - rsa.misc.disposition: $~log.rsa.misc.disposition
+    - rsa.misc.filter: $~log.rsa.misc.filter
+    - rsa.email.subject: $~log.rsa.email.subject
+    - rsa.email.subject: $~log.rsa.misc.event_type
+    - rsa.internal.event_desc: $~log.rsa.internal.event_desc
+    - rsa.misc.checksum: $~log.rsa.misc.checksum
+  - check:
+      - rsa.internal.event_desc: +ef_exists
+    logpar:
+      - rsa.internal.event_desc: >-
+          <~log.fill> to \<\<<~log.rsa.email.email_dst>>; reason:<~log.rsa.misc.result>; <~log.fill>
+      - rsa.internal.event_desc: >-
+          to=\<\<<~log.rsa.email.email_dst>>, delay=<~log.fill>, xdelay=<~log.fill>, mailer=<network.protocol>,
+          pri=<~log.fill>, relay=<host.hostname>[<source.ip>], <~log.fill>
+      - rsa.internal.event_desc: >-
+          to=\<\<<~log.rsa.email.email_dst>, delay=<~log.fill>, xdelay=<~log.fill>, mailer=<network.protocol>,
+          pri=<~log.fill>, relay=<host.hostname>
+    map:
+      - rsa.misc.result: $~log.rsa.misc.result
+      - rsa.email.email_dst: $~log.rsa.email.email_dst
+      - rsa.internal.event_desc: +ef_delete
+  - check:
+      - rsa.web.fqdn: +ef_exists
+    map:
+      - related.hosts: +a_append/$rsa.web.fqdn
+  - check:
+      - host.hostname: +ef_exists
+    map:
+      - related.hosts: +a_append/$host.hostname
+  - check:
+      - server.domain: +ef_exists
+    map:
+      - related.hosts: +a_append/$server.domain
+  - check:
+      - destination.ip: +ef_exists
+    map:
+      - related.ip: +a_append/$destination.ip
+  - check:
+      - source.ip: +ef_exists
+    map:
+      - related.ip: +a_append/$source.ip
+  - map:
```

Comment:
add new empty line before to improve readability.

Reply:
Done